### PR TITLE
Improved simplified prediction output

### PR DESCRIPTION
**Pull request by kevinykuo**
- The `simplify` argument in `predict()` and `evaluate()` now take only a boolean value (as opposed to also accepting a user-defined function). This simplifies the `predict()` and `evaluate()` interfaces, and users can still perform post-processing outside of these functions.
- For prediction, when `simplify = TRUE` (default), the individual arrays returned by TF are endorsed with a `tf_prediction` type, which allows us to control printing behavior in a `tibble` via `type_sum.tf_prediction()`. The default output now looks like the following:
```
model %>% predict(mtcars_input_fn(test_data), simplify = TRUE)
# A tibble: 7 x 5
  probabilities  logits classes class_ids   logistic
         <list>  <list>  <list>    <list>     <list>
1 <1, 6.69e-14> <-30.3>     <0>       <0> <6.69e-14>
2 <1, 2.18e-13> <-29.2>     <0>       <0> <2.18e-13>
3 <1, 6.86e-16> <-34.9>     <0>       <0> <6.86e-16>
4 <1, 1.84e-25>   <-57>     <0>       <0> <1.84e-25>
5        <1, 0> <-93.8>     <0>       <0>        <0>
6        <1, 0> <-89.8>     <0>       <0>        <0>
7 <1, 6.53e-28> <-62.6>     <0>       <0> <6.53e-28>
```
- The output for `evaluate()` remains unchanged.
- The tutorial for [writing custom simplifiers](https://tensorflow.rstudio.com/tfestimators/articles/simplify_function.html) has been removed.

### CODE REVIEW

**Comment by kevinykuo** [comment]
cc @terrytangyuan @edgararuiz @kevinushey

File: R/simplify_fn.R [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
+  else
+    paste0(x)
+}
+
+simple_simplify_predictions_fn <- function(results) {
+  results %>%
+    purrr::map(~ purrr::map(.x, list)) %>% 
+    purrr::transpose() %>%
+    purrr::map(~ purrr::map(.x, 
+                            purrr::compose(as_tf_prediction, unlist))
```
Comment (terrytangyuan) [inline R/simplify_fn.R]: Looks good to me! But would multiple `map`s here affect the performance?

Comment (kevinykuo) [reply R/simplify_fn.R]: Good point! We can probably avoid the `list()`-`unlist()` here; lemme give it a shot.

**Comment by jjallaire** [comment]
LGTM as well, feel free to merge when ready!

**Pull request merged**